cSploit/src/org/csploit/android/MainActivity.java

```diff
@@ -139,15 +145,16 @@ private void createUpdateLayout() {
         lv.setVisibility(View.GONE);
         findViewById(R.id.textView).setVisibility(View.GONE);
 
-        createUpdateStatusText();
-
-        mUpdateStatus
-                .setText(UPDATE_MESSAGE.replace("#STATUS#", "..."));
+        mUpdateStatus.setVisibility(View.VISIBLE);
```

Comment:
I moved the initialization of mUpdateStatus to onCreate(). Since it's called only once on the activity lifetime, it shouldn't be null, and it's not instantiated every time we change a layout (although there aren't so much changes...).

btw, there's a null check in onCreate(), because onCreate() was called after coming back from selecting a wifi network. Since I created a new method, init(), for common initializations and checks, that null check is not necessary anymore.